packages/lodestar/src/sync/reqResp/reqResp.ts

```diff
@@ -97,7 +99,31 @@ export class SyncReqResp implements ISyncReqResp {
     } catch (e) {
       this.network.reqResp.sendResponse(id, e, null);
     }
-    // TODO handle incorrect forkVersion or disjoint finalizedCheckpoint
+    if (await this.shouldDisconnectOnHello(peerInfo, request)) {
+      this.network.reqResp.goodbye(peerInfo, 0n);
+    }
+  }
+
+  public async shouldDisconnectOnHello(peerInfo: PeerInfo, request: Hello): Promise<boolean> {
+    const headBlock = await this.db.block.getChainHead();
+    const state = await this.db.state.get(headBlock.stateRoot);
+    const fork = state.fork;
```

Comment:
I think chain module should expose getFork method

Reply:
I also need state for subsequence calls, the new code looks better anyway.